### DISCUSSION TITLE
Using enums for permission

### DISCUSSION THREAD

**Discussion by berkinanik** [epicweb-dev/epic-stack#422] [Q&A] [answered]
In permission model, `action`:`entity`:`access` fields are string and I know sqlite does not support enums. _btw current implementation is nice imo for both typescript support and intellisense._ 
https://github.com/epicweb-dev/epic-stack/blob/4fd50cd440d1863d4cc58b843f0871472dfeb4fe/app/utils/permissions.ts#L62-L66
I was just wondering if we could benefit from enums for these fields with postgres etc. And how would the permission parsing be implemented that way.
```prisma
// File: schema.prisma
enum Action {
  CREATE
  READ
  // ..
}
```

PS: I was first thinking if there was a way to create `Access` type dynamically, if access types start to grow for example.
https://github.com/epicweb-dev/epic-stack/blob/4fd50cd440d1863d4cc58b843f0871472dfeb4fe/app/utils/permissions.ts#L64-L64
Like:
```typescript
type Access = AllPossibleUniqueStringCombinations<'own' | 'team' | 'invited' | 'any'>
// would infer all possible string combinations
```
But ended up thinking about why those weren't enums in the schema earlier

**Comment by sergiodxa** [comment, marked as answer]
EpicStack uses SQLite, not PostgreSQL, and AFAIK SQLite doesn't support DB-level enums, so the only option is to have them in the code and not the schema.
